src/array.php

```diff
@@ -120,3 +120,25 @@ function tail($list)
     $list = to_array($list);
     return array_pop($list);
 }
+
+/**
+ * Index an array by a key.
+ *
+ * @param array|Traversable $source
```

Comment:
I think this would be more descriptive (to help the user know they can pass in objects)
`@param array|object[]|Traversable $source`